### PR TITLE
light: correctly handle contexts (backport -> v0.34.x)

### PR DESCRIPTION
**Pull request by cmwaters**
This PR allows cancelling contexts or using deadlines to actually work. When state sync was using the light client there were occasions where getting the AppHash was taking longer than 10 seconds and so the deadline was getting exceeded but because the light client didn't handle this error properly, it would actually get stuck.

### CODE REVIEW

File: statesync/syncer.go [hunk 1]
```diff
@@ -256,7 +260,7 @@ func (s *syncer) Sync(snapshot *snapshot, chunks *chunkQueue) (sm.State, *types.
 		go s.fetchChunks(ctx, snapshot, chunks)
 	}
 
-	pctx, pcancel := context.WithTimeout(context.Background(), 10*time.Second)
+	pctx, pcancel := context.WithTimeout(context.Background(), 15*time.Second)
```
Comment (marbar3778) [inline statesync/syncer.go:263]: could we add a comment for why this time was chosen

Comment (cmwaters) [reply statesync/syncer.go:263]: Well the motivation for the change was that I was seeing deadline errors with the e2e tests but apart from that the decision is quite arbitrary

Comment (marbar3778) [reply statesync/syncer.go:263]: does this timeout effect live networks?

Comment (cmwaters) [reply statesync/syncer.go:263]: what exactly do you mean by that?

Comment (marbar3778) [reply statesync/syncer.go:263]: Do we see this timeout accidentally being exceeded on main nets?

Comment (cmwaters) [reply statesync/syncer.go:263]: I wouldn't know

**Review by tychoish — APPROVED** [review]

**Pull request merged**